TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/LaneNode.cs

```diff
@@ -9,21 +9,22 @@ public class LaneNode : Node<LaneNode>
         private RoadNode _roadNode;
         private Vehicle _vehicle;
         /// <summary>Creates a new isolated lane node without any previous or next nodes</summary>
-        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode) : this(position, rotation, roadNode, null, null){}
+        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, float distanceToNextNode) : this(position, rotation, roadNode, null, null, distanceToNextNode){}
         
         /// <summary>Creates a new lane node</summary>
         /// <param name="position">The position of the node</param>
         /// <param name="rotation">The rotation of the node</param>
         /// <param name="roadNode">The road node this lane node relates to</param>
         /// <param name="prev">The previous lane node. Pass `null` if there is no previous</param>
         /// <param name="next">The next lane node. Pass `null` if there is no next</param>
-        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, LaneNode prev, LaneNode next)
+        public LaneNode(Vector3 position, Quaternion rotation, RoadNode roadNode, LaneNode prev, LaneNode next, float distanceToNextNode)
```

Comment:
The new variable "distanceToNextNode" seems to be used in a backwards way when building the list in Lane.cs (see other comment). Maybe rename to "distanceToPreviousNode" ?

Reply:
I can switch the logic to be distance to previous node instead, might make more sense